File: 003.py
number = 600851475143 

def isPrime(number):
    for i in range(2, int(number ** 0.5) + 1):
        if(number % i == 0):
            return False
    return True

def primeFactor(number):
    for i in range(2, int(number ** 0.5 + 1)):
        if(number % i == 0) and isPrime(i):
            while(number % i == 0):
                #print(i, number)
                number /=i
            print(i)

    if(number != 1):
        print(int(number))
                
primeFactor(number)

File: C/003.c
#include <stdio.h>
#include <math.h>

#define TRUE 1
#define FALSE 0

int isPrime(long long *n);
void primeFactors(long long *n);

int main(void)
{
    long long number = 600851475143;

    primeFactors(&number);
}

int isPrime(long long *n)
{
    for (long i = 2, sq = sqrt(*n); i < sq + 1; i++)
        if (*n % i == 0)
            return FALSE;
    return TRUE;
}

void primeFactors(long long *n)
{
    for (long long i = 2, sq = sqrt(*n); i < sq + 1; i++)
    {
        if (*n % i == 0 && isPrime(&i) == TRUE)
        {
            while (*n % i == 0)
                *n /= i;
            printf("%lld\n", i);
        }
    }
    if (*n != 1)
        printf("%lld", *n);
}
